### PR TITLE
V3 API Mediation Layer related cleanup

### PR DESCRIPTION
**Pull request by balhar-jakub**
Describe your pull request here:
I went through the API ML related doc for V3 and cleansed pieces that aren't relevant for V3 or claim information such as supported from specific Zowe version. 

List the file(s) included in this PR:

After creating the PR, follow the instructions in the comments.

### CODE REVIEW

File: docs/appendix/zowe-yaml-configuration.md [hunk 1]
```diff
@@ -449,6 +449,32 @@ These configurations can be used under the `components.gateway` section:
  Specifies the maximum connections for each service.
 - **`apiml.server.maxTotalConnections`**  
  Specifies the total connections for all services registered under API Mediation Layer.
+- **`apiml.security.oidc.enabled`**  
+  Specifies the global feature toggle. Set the value to `true` to enable OIDC authentication functionality.
+
+- **`apiml.security.oidc.registry`**  
+  Specifies the SAF registry used to group the identities recognized as having a OIDC identity mapping. The registry name is the string used during the creation of the mapping between the dustributed and mainframe user identities. For more information, see the [ESM configuration](#esm-configuration).
```
Comment (anaxceron) [inline docs/appendix/zowe-yaml-configuration.md:456]: should be "... **an** OIDC ..."

File: docs/appendix/zowe-yaml-configuration.md [hunk 2]
```diff
@@ -449,6 +449,32 @@ These configurations can be used under the `components.gateway` section:
...
+- **`apiml.security.oidc.jwks.uri`**  
+  Specifies the URI obtained from the authorization server's metadata where the Gateway will query for the JWK used to sign and verify the access tokens.
+
+- **`apiml.security.oidc.jwks.refreshInternalHours`**  
+  Specifies the frequency in hours to refresh the JWK keys from the OIDC provider. Defaults to one hour.
+
+- **`apiml.security.oidc.identityMapperUser`**  
+  (Optional) If the userId is different from the default Zowe runtime userId (`ZWESVUSR`), specify the `identityMapperUser` userId to configure API ML access to the external user identity mapper.
+
+  **Note:** User authorization is required to use the `IRR.RUSERMAP` resource within the `FACILITY` class. The default value is `ZWESVUSR`. Permissions are set up during installation with the `ZWESECUR` JCL or workflow. To authenticate to the mapping API, a JWT is sent with the request. The token represents the user that is configured with this property.
```
Comment (anaxceron) [inline docs/appendix/zowe-yaml-configuration.md:467]: Please use the Docusaurus admonition syntax:
https://docusaurus.io/docs/next/markdown-features/admonitions

File: docs/appendix/zowe-yaml-configuration.md [hunk 3]
```diff
@@ -512,6 +538,18 @@ These configurations can be used under the `components.caching-service` section:
  Specifies eviction strategy to be used when the storage size is achieved.
 - **`storage.vsam.name`**  
  Specifies the data set name of the caching service VSAM data set.
+- **`storage.infinispan.initialHosts`**
+
+  This property specifies the list of cluster nodes (members). In case of multiple instances, the value for each Caching Service instance can be either a list of all the members, separated by a comma, or just the replica. The format is `${haInstance.hostname}[${zowe.components.caching-service.storage.i˚nfinispan.jgroups.port}]`.  either a list of all the members, separated by a comma, or just the replica. The format is `${haInstance.hostname}[${zowe.components.caching-service.storage.infinispan.jgroups.port}]`.
```
Comment (anaxceron) [inline docs/appendix/zowe-yaml-configuration.md:543]: The phrase "either a list of all the members, separated by a comma ... " is repeated twice in this definition. It looks like a copy/paste error. Can you confirm?

File: docs/appendix/zowe-yaml-configuration.md [hunk 4]
```diff
@@ -512,6 +538,18 @@ These configurations can be used under the `components.caching-service` section:
...
+
+- **`storage.infinispan.persistence.dataLocation`**
+
+  The path where the Soft-Index store keeps its data files for the Infinispan Soft-Index Cache Store.
+  The default value is `data`. If you run the Caching Service in HA and the instances use the same filesystem, you have to specify a different value of the `CACHING_STORAGE_INFINISPAN_PERSISTENCE_DATALOCATION` property for each instance. For more information, see the [Soft-Index File Store](https://infinispan.org/blog/2014/10/31/soft-index-file-store).
```
Comment (anaxceron) [inline docs/appendix/zowe-yaml-configuration.md:548]: We should spell out "HA."

File: docs/appendix/zowe-yaml-configuration.md [hunk 5]
```diff
@@ -512,6 +538,18 @@ These configurations can be used under the `components.caching-service` section:
...
+
+- **`storage.infinispan.jgroups.port`**
+
+  The port number used by Infinispan to synchronise data among cahing-service instances.
```
Comment (anaxceron) [inline docs/appendix/zowe-yaml-configuration.md:552]: typo on "cahing-service"

File: docs/getting-started/overview.md [hunk 6]
```diff
@@ -79,32 +75,39 @@ The following diagram illustrates the single point of access through the Gateway
...
 **Caching Service**
 
-An API is provided in high-availability mode which offers the possibility to store, retrieve, and delete data associated with keys. The service can only be used by internal Zowe services and is not exposed to the internet.
+An API is provided in high-availability mode which offers the possibility to store, retrieve, and delete data associated 
+with keys. 
 
 #### Onboarding APIs
-Essential to the API Mediation Layer ecosystem is the API services that expose their useful APIs. Use the following topics to discover more about adding new APIs to the API Mediation Layer and using the API Catalog:
+Essential to the API Mediation Layer ecosystem is the API services that expose their useful APIs. Use the following topics 
+to discover more about adding new APIs to the API Mediation Layer and using the API Catalog:
```
Comment (anaxceron) [inline docs/getting-started/overview.md:110]: Please change "discover" to "learn." "Discover" sounds a little awkward in this particular use.

File: docs/getting-started/zowe-security-overview.md [hunk 7]
```diff
@@ -29,8 +29,8 @@ A certificate contains an identity (a hostname, or an organization, or an indivi
 A certificate can be self-signed or issued by a Certificate Authority (CA). A CA is a trusted organization which provides infrastructure for creation, validation and revocation of the certificates according to the contemporary security standards.
 
 **Note:** 
-For testing purposes of Zowe, it is acceptable to use certificates issued and signed either by the company's local CA, or even self-signed certificates issued by Zowe security tools specific for the target technology platform.
-Use of self-signed certificates, however, is not recommended for production environments.
+For testing purposes of Zowe, it is acceptable to use certificates issued and signed either by the company's local CA, or even certificates issued by Zowe security tools and signed by generated CA specific for the target technology platform.
```
Comment (anaxceron) [inline docs/getting-started/zowe-security-overview.md:32]: - Please use the Docusaurus syntax for admonitions. See previous statement for the URL.
- We should spell out "CA" on first use.

File: docs/troubleshoot/troubleshoot-apiml.md [hunk 8]
```diff
@@ -352,7 +352,7 @@ main¨ o.a.http.impl.client.DefaultHttpClient   : Retrying connect to {s}->https
 
 The Zowe started task needs to run under a user with sufficient privileges. As a workaround, you can try to run the started task under the same user ID as z/OSMF (typically IZUSVR).
 
-The hostname that is displayed in the details of the exception is a valid hostname. You can validate that the hostname is valid by using `ping` command on the same mainframe system. For example, `ping USILCA32.lvn.broadcom.net`. If it is valid, then the problem can be caused by insufficient privileges of your started task that is not allowed to do network access.
+The hostname that is displayed in the details of the exception is a valid hostname. You can validate that the hostname is valid by using `ping` command on the same mainframe system. For example, `ping system.lvn.broadcom.net`. If it is valid, then the problem can be caused by insufficient privileges of your started task that is not allowed to do network access.
```
Comment (anaxceron) [inline docs/troubleshoot/troubleshoot-apiml.md:355]: "that is not allowed to **do** network access" sounds awkward. can we change it to "that is not allowed network access"?

File: docs/contribute/guidelines-code/categories.md [hunk 9]
```diff
@@ -17,7 +17,7 @@ For each area of the codebase, there are established and favored programming lan
 - **CLI** - Node.js, TypeScript
 - **Desktop UI** - Node.js, JavaScript
 - **APIs** - C, Assembler, Java, Spring
-- **API Mediation Layer** - Java, Spring
+- **API Mediation Layer** - Java, Spring, JavaScript
```
Comment (anaxceron) [inline docs/contribute/guidelines-code/categories.md:20]: Please use the Docusaurus admonition syntax:
https://docusaurus.io/docs/next/markdown-features/admonitions

**Review by anaxceron — CHANGES_REQUESTED** [review]
Looks pretty good @balhar-jakub. Made some comments that should be addressed.

**Comment by balhar-jakub** [comment]
@anaxceron Resolved all the comments.

**Review by anaxceron — APPROVED** [review]

File: docs/user-guide/api-mediation/api-mediation-caching-service.md [hunk 10]
```diff
@@ -1,13 +1,14 @@
...
-As an API developer, you can use the Caching Service as a storage solution to enable resource sharing between service instances, thereby ensuring High Availability of services. The Caching Service makes it possible to store, retrieve, and delete data associated with keys. The Caching Service is designed to make resource sharing possible for services that cannot be made stateless in two ways:
+As an API developer, you can use the Caching Service as a storage solution to enable resource sharing between service instances, thereby ensuring High Availability of services. The Caching Service makes it possible to store, retrieve, and delete data associated with keys. The Caching Service is designed to make resource sharing possible for services that cannot be made stateless by using following backends:
 
-- Using VSAM to store key/value pairs for production
-
-- Using InMemory
+- Using Infinispan that is part of Caching Service
+- Using Redis running off-platform
+- {Deprecated} Using VSAM
+- {Development Use Only} Using InMemory
```
Comment (t1m0thyj) [inline docs/user-guide/api-mediation/api-mediation-caching-service.md:8]: In Docusaurus V3, curly braces need to be escaped:
```suggestion
- \{Deprecated\} Using VSAM
- \{Development Use Only\} Using InMemory
```

Escaping them should fix the build error 🙏 
image

**Pull request merged**